### DISCUSSION TITLE
private / protected properties and mounting

### DISCUSSION THREAD

**Discussion by EmranMR** [livewire/livewire#8726] [1. Help] [answered]
Hello!
I am a bit confused as I could not find any information on the documentation nor here? I would super appreciate if anyone let me know the Livewire flow under these circumstances 
imagine a construct like this

```php
//...

private Carbon $carbon;


public function mount()
{
    $this->carbonDate = Date::now();
    //...
}

public function updatedSomething($thatThing)
{
   $this->carbon->setWhatever($thatThing);
}

// ...
```

Now when the property updates on front end and setting the `private carbon` in the function I get something in lines of 

```
$carbon must not be accessed before initialization
```

### The Questions:
1. Do private and public properties get cleared every time the component is re-rendered?
3. Is the workflow something like:
```
ComponentCreates -> Json Payload -> frontEnd do stuff -> JsonPayload back -> use the payload props to set the public properties? 
```
##### The docs say
> Livewire components don't use __construct() because Livewire components are re-constructed on subsequent network requests, and we only want to initialize the component once when it is first created.
- So basically if the component is initialized once, why do the private and protected properties get cleared after each request upon mounting 🤷‍♂️ 

- For now I use `#[Locked]` anyways, as it is fit for purpose and sort of replicates private, not allowing front end modifying the property, but would be nice to know the reason for my own information.

**Comment by PhiloNL** [comment, marked as answer]
Hi @EmranMR,

> Do private and public properties get cleared every time the component is re-rendered?
> So basically if the component is initialized once, why do the private and protected properties get cleared after each request upon mounting 🤷‍♂️

The state of a component is stored on the client-side and Livewire only stores public properties. For this reason, protected and private properties won't work on consecutive requests, as these values are never exposed to the client side. For example, if you have a private property called `password` it would never be private if the state was still sent to the client side given it could be extracted from the response.

Behind the scenes Livewire uses Synthesizers, you can find a [full explanation here](https://livewire.laravel.com/docs/synthesizers#understanding-synthesizers).

So using `#[Locked]` is the way to go.

**Reply by EmranMR** [reply]
@PhiloNL  Thank you! I really appreciate your detailed answer! I shall indeed look at the `Synthesizers ` for more information on Livewire magics!

**Reply by PhiloNL** [reply]
@EmranMR no problem. I made a little package recently to lock your properties by default in case you are using `#[Locked]` a lot:
https://github.com/wire-elements/livewire-strict

**Reply by EmranMR** [reply]
Ah perfect! exactly what I was looking for!
